src/ui/viewmodels/MessageBoxViewModel.hh

```diff
@@ -0,0 +1,167 @@
+#ifndef RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#define RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#pragma once
+
+#include "ui/WindowViewModelBase.hh"
+
+namespace ra {
+namespace ui {
+namespace viewmodels {
+
+enum class MessageBoxButtons
+{
+    OK,
+    OKCancel,
+    YesNo,
+    YesNoCancel,
+    RetryCancel,
+};
+
+enum class MessageBoxIcon
+{
+    None,
+    Info,
+    Warning,
+    Error,
+};
+
+class MessageBoxViewModel : public WindowViewModelBase
+{
+public:
+    MessageBoxViewModel() noexcept {}
+
+    MessageBoxViewModel(const std::wstring& sMessage) noexcept
```

Comment:
Mark this as `explicit `to prevent implicit conversion.
And I think it would be better to give some default parameters such as `::GetActiveWindow` for `HWND `and `MessageBoxIcon`::Info for Icon, and `MessageBoxButtons::OK` for the button. Those are just examples, if there are better defaults that you can think of then be my guest.
I think the "specific" message boxes should be subclassed from this one but if you don't want to do it I guess that's fine too.